### ISSUE TITLE
blade mail return damage to muerta(ult form) even if his target is magic immunity.

### ISSUE THREAD

**Issue by Misky11** [ValveSoftware/Dota2-Gameplay#7747]
### Item name

blade mail

### Description

hello i was testing demo as muerta then i press ultimate and attacked a target that activated both bkb and blade mail on it and saw muerta's attacks doesnt damage but blade mail returns damage to her.
then i test jakiro's ultimate(without aghanim) or lich's ultimate and saw damage doesnt return to them.
![Uploading 20230326231746_1.jpg…]()
![Uploading 20230326231925_1.jpg…]()
![Uploading 20230326231757_1.jpg…]()

### Example Match ID and Timestamp

demo

**Comment by Aluerie** [comment]
this is not really a bug; 

there is a difference between not doing damage (lich/jakiro) to bkb-targets and doing damage that is getting reduced to 0 (muerta). and since BM returns damage pre-reduction -> muerta gets full damage back as intended while jakiro/lich do not.

even tho one can say it's just a difference of semantics so maybe valve can/should change that.

**Comment by Misky11** [comment]
> this is not really a bug;
> 
> there is a difference between not doing damage (lich/jakiro) to bkb-targets and doing no damage (muerta). and since BM returns damage pre-reduction -> muerta gets full damage back as intended while jakiro/lich do not.
> 
> even tho one can say it's just a difference of semantics so maybe valve can/should change that.

just imagine this you are a muerta (your bkb is on cooldown) and you are hitting a target without bkb but your gunslinger is targeting a target activitaded both bkb and blade mail... you will got extra damage and die in 5-6seconds at most.

**Comment by TheChetu** [comment]
> this is not really a bug;
> 
> there is a difference between not doing damage (lich/jakiro) to bkb-targets and doing no damage (muerta). and since BM returns damage pre-reduction -> muerta gets full damage back as intended while jakiro/lich do not.
> 
> even tho one can say it's just a difference of semantics so maybe valve can/should change that...
> 
> but this is a bug tracker - not a place for balance suggestions.

This may not be a bug but I would argue that it is a mistake in the implementation unless valve says this is intended. Your interpretation is correct and the way it is currently implemented is that the incoming damage is blocked at 100% effectiveness rather than being reduced to zero so the full damage is still considered for the Blade Mail despite there being no actual damage taken. 

I personally encountered this last night against a Blade Mail Axe with BKB, he would taunt me and I would kill myself in two shots. Doesn't really make sense since he took zero damage and there is no easy way to prevent this unless I already had my BKB on or was able to use it just before a taunt.

**Comment by richterlevania3** [comment]
Being mechanically correct or not, it just doesn't make sense. I'm not aware of any other interaction where I take damage when dealing magic damage to someone with BKB and BM.

Since other magic damage physical attacks can't even target someone with BKB (Drow's Frost Arrows, Enchantress's Impetus e the like), I propose Muerta not be able to target a hero with an active BKB.

**Comment by Misky11** [comment]
after patch 7.33 still it has some problem and that is... they totally changed spell immunity(it gives magic resistance) if enemy use blade mail on bkb and you as muerta hit still you recieve full damage i dont know its intended or not but when you are hitting a target that recieve 100 damage and you recieve back 250 damage is not really balanced

can really someone from valve respond to this?

**Comment by TheChetu** [comment]
> after patch 7.33 still it has some problem and that is... they totally changed spell immunity(it gives magic resistance) if enemy use blade mail on bkb and you as muerta hit still you recieve full damage i dont know its intended or not but when you are hitting a target that recieve 100 damage and you recieve back 250 damage is not really balanced
> 
> can really someone from valve respond to this?
> 
> ![20230422100451_1](https://user-images.githubusercontent.com/107326195/233765106-bed00d0e-29eb-4b19-bbb0-00e8113c0125.jpg) ![20230422100445_1](https://user-images.githubusercontent.com/107326195/233765114-f9551246-3992-4374-8c23-21e2b3e01bc5.jpg)

This is correct, the damage you do is reduced by 50% and the damage they do is calculated before reductions so returned at 105%. As BKB and spell immunity in general has been changed the damage from Muerta's ult should be working properly.

For example, if an Axe has 75% physical damage reduction from Berserker's Call and his armor stacks/items and a Phantom Assassin were to crit the Axe for 1500 damage before reductions the Blade Mail would return 1500 physical damage to the Phantom Assassin and the Axe would take the reduced damage of 375.
